src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs

```diff
@@ -115,6 +119,90 @@ public void InitializeService(ServiceHost serviceHost)
             await HandleGetTableOrViewRequest(metadataParams, "view", requestContext);
         }
 
+        /// <summary>
+        /// Handles the request for generating all server metadata
+        /// </summary>
+        internal static async Task HandleGenerateServerTableMetadataRequest(GenerateServerTableMetadataParams metadataParams,
+            RequestContext<bool> requestContext)
+        {
+            MetadataService.ConnectionServiceInstance.TryFindConnection(metadataParams.OwnerUri, out ConnectionInfo connectionInfo);
+
+            if (connectionInfo != null)
+            {
+                using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connectionInfo, "metadata"))
+                {
+                    try
+                    {
+                        if (MetadataScriptTempFileStream.IsScriptTempFileValid(connectionInfo.ConnectionDetails.ServerName))
+                        {
+                            await requestContext.SendResult(true);
```

Comment:
is this true result sent here used by ADS? How does ADS differentiate it from the result sent on line 147?

Reply:
ADS doesn't use the returned bool value. I couldn't find a way to specify `RequestContext<void>` or just `RequestContext`  since what's being returned isn't overly useful, so ended up going with `RequestContext<bool>`.

Reply:
If you're not waiting on a result then this should probably be a notification, not a request (which doesn't have a response)

Reply:
or can you just return the file path with the second result? And make a new result type that has the string path and success

Reply:
@Charles-Gagnon, to return a notification I would need to swap out `SetRequestHandler` for `SetEventHandler`, and then I wouldn't need to return anything or would I still need to invoke `sendEvent` on the `serviceHost`?

@kisantia, would the success flag really be needed since sendError is being used in the handle request endpoint when there is an error? That part seems kind of redundant, but the file path string makes sense to me.

Reply:
Correct and correct

> @Charles-Gagnon, to return a notification I would need to swap out `SetRequestHandler` for `SetEventHandler`, and then I wouldn't need to return anything or would I still need to invoke `sendEvent` on the `serviceHost`?